presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardOrganizerUtil.java

```diff
@@ -0,0 +1,190 @@
+            new ColumnInfo(3, VARCHAR));
+
+    private DBI dbi;
+    private Handle dummyHandle;
+    private File dataDir;
+    private ShardManager shardManager;
+    private MetadataDao metadataDao;
+
+    @BeforeMethod
+    public void setup()
+    {
+        dbi = new DBI("jdbc:h2:mem:test" + System.nanoTime());
+        dummyHandle = dbi.open();
+        dataDir = Files.createTempDir();
+        metadataDao = dbi.onDemand(MetadataDao.class);
+        shardManager = createShardManager(dbi);
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void teardown()
+    {
+        dummyHandle.close();
+        FileUtils.deleteRecursively(dataDir);
+    }
+
+    @Test
+    public void testShardIndexInfoConverter()
+            throws Exception
+    {
+        long day1 = Duration.ofDays(Duration.ofNanos(System.nanoTime()).toDays()).toMillis();
+        long day2 = Duration.ofDays(Duration.ofMillis(day1).toDays() + 1).toMillis();
+
+        long tableId = createTable("test", columns);
+        metadataDao.updateTemporalColumnId(tableId, 1L);
+        Table tableInfo = metadataDao.getTableInformation(tableId);
+        metadataDao.updateTemporalColumnId(tableId, 1L);
+
+        List<ShardInfo> shards = ImmutableList.<ShardInfo>builder()
+                .add(shardInfo(
+                        UUID.randomUUID(),
+                        "node1",
+                        ImmutableList.of(new ColumnStats(1, day1, day1 + 10), new ColumnStats(2, 13L, 14L), new ColumnStats(3, "aaa", "abc"))))
+                .add(shardInfo(
+                        UUID.randomUUID(),
+                        "node1",
+                        ImmutableList.of(new ColumnStats(1, day2, day2 + 100), new ColumnStats(2, 2L, 20L), new ColumnStats(3, "aaa", "abc"))))
+                .add(shardInfo(
+                        UUID.randomUUID(),
+                        "node1",
+                        ImmutableList.of(new ColumnStats(1, day1, day2), new ColumnStats(2, 2L, 11L), new ColumnStats(3, "aaa", "abc"))))
+                .build();
+
+        long transactionId = shardManager.beginTransaction();
+        shardManager.commitShards(transactionId, tableId, columns, shards, Optional.empty());
+        Set<ShardMetadata> shardMetadatas = shardManager.getNodeShards("node1");
+
+        Set<ShardIndexInfo> expected = shardInfoToShardIndexInfo(tableInfo, shards, false);
+        Set<ShardIndexInfo> actual = ImmutableSet.copyOf((toShardIndexInfo(dbi, metadataDao, tableInfo, shardMetadatas, false)));
+        assertEquals(actual, expected);
+
+        Set<ShardIndexInfo> expectedSortRange = shardInfoToShardIndexInfo(tableInfo, shards, true);
+        Set<ShardIndexInfo> actualSortRange = ImmutableSet.copyOf((toShardIndexInfo(dbi, metadataDao, tableInfo, shardMetadatas, true)));
+        assertEquals(actualSortRange, expectedSortRange);
+    }
+
+    private Set<ShardIndexInfo> shardInfoToShardIndexInfo(Table tableInfo, Collection<ShardInfo> shards, boolean includeSort)
```

Comment:
This is rather complex code for a test and warrants its own tests. It also makes it hard to see what is being tested. I would prefer to just create the expected objects manually (will be a bit long but much easier to follow).